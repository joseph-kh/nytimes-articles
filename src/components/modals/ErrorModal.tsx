import React from "react";
import { Modal, StyleSheet, View, TouchableOpacity } from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import BoldText from "@/components/ui/texts/BoldText";
import SemiBoldText from "@/components/ui/texts/SemiBoldText";
import RegularText from "@/components/ui/texts/RegularText";
import Divider from "@/components/ui/Divider";
import Center from "@/components/ui/containers/Center";
import COLORS from "@/theme/colors";
import { horizontalScale, moderateScale, verticalScale } from "@/utils/scaler";

type Props = {
  isVisible: boolean;
  title: string;
  message: string;
  ctaText: string;
  toggleModal: () => void;
  onConfirm: () => void;
};

const ErrorModal: React.FC<Props> = (props) => {
  const { isVisible, title, message, ctaText, toggleModal, onConfirm } = props;

  return (
    <Modal
      animationType="slide"
      transparent={true}
      visible={isVisible}
      onRequestClose={toggleModal}
    >
      <Center style={styles.container}>
        <View style={styles.innercontainerStyle}>
          <View style={styles.topContainerStyle}>
            <MaterialCommunityIcons
              name="close-circle"
              size={moderateScale(42)}
              color={COLORS.error}
              style={styles.iconStyle}
            />
            <BoldText style={styles.titleTextStyle}>{title}</BoldText>
            <RegularText style={styles.messageTextStyle}>{message}</RegularText>
          </View>
          <Divider style={styles.dividerStyle} />
          <TouchableOpacity onPress={onConfirm}>
            <SemiBoldText style={styles.ctaTextStyle}>{ctaText}</SemiBoldText>
          </TouchableOpacity>
        </View>
      </Center>
    </Modal>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },

  innercontainerStyle: {
    borderRadius: moderateScale(8),
    paddingVertical: verticalScale(15),
    backgroundColor: COLORS.white,
    elevation: 5,
    shadowRadius: 4,
    shadowColor: "#000",
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.25,
  },

  topContainerStyle: {
    paddingHorizontal: horizontalScale(30),
  },

  iconStyle: {
    alignSelf: "center",
  },

  titleTextStyle: {
    textAlign: "center",
  },

  messageTextStyle: {
    textAlign: "center",
  },

  dividerStyle: {
    marginVertical: verticalScale(10),
  },

  ctaTextStyle: {
    textAlign: "center",
    color: COLORS.primary,
    fontSize: moderateScale(16),
    alignSelf: "center",
  },
});

export default ErrorModal;
