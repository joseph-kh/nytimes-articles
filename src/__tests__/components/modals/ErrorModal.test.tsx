import React from "react";
import renderer from "react-test-renderer";
import ErrorModal from "@/components/modals/ErrorModal";

describe("ErrorModal Component", () => {
  it("renders correctly when visible", () => {
    const tree = renderer
      .create(
        <ErrorModal
          isVisible={true}
          title="Error Title"
          message="Error Message"
          ctaText="Confirm"
          toggleModal={() => {}}
          onConfirm={() => {}}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when not visible", () => {
    const tree = renderer
      .create(
        <ErrorModal
          isVisible={false}
          title="Error Title"
          message="Error Message"
          ctaText="Confirm"
          toggleModal={() => {}}
          onConfirm={() => {}}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
