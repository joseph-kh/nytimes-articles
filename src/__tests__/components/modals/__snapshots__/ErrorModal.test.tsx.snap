// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal Component renders correctly when not visible 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
/>
`;

exports[`ErrorModal Component renders correctly when visible 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 12,
          "elevation": 5,
          "paddingVertical": 24.642857142857142,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
        }
      }
    >
      <View
        style={
          {
            "paddingHorizontal": 60,
          }
        }
      >
        <Text />
        <Text
          style={
            {
              "fontFamily": "Bold",
              "fontSize": 21,
              "textAlign": "center",
            }
          }
        >
          Error Title
        </Text>
        <Text
          style={
            {
              "fontFamily": "Regular",
              "fontSize": 21,
              "textAlign": "center",
            }
          }
        >
          Error Message
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "#7e7e7e",
            "borderBottomWidth": 0.75,
            "marginVertical": 16.428571428571427,
            "opacity": 0.4,
          }
        }
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "alignSelf": "center",
              "color": "#51e2c3",
              "fontFamily": "SemiBold",
              "fontSize": 24,
              "textAlign": "center",
            }
          }
        >
          Confirm
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
